File: BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.go
func maxProfit(prices []int) int {
    max := 0
    
    min := prices[0]
    for i := 1; i < len(prices); i++ {
        tmp := prices[i] - min
        if tmp > max {
            max = tmp
        }
        if prices[i] < min {
            min = prices[i]
        }
    }
    
    return max
}

File: BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        bestBuy = prices[0]
        bestProfit = 0
        
        for i in range(1, len(prices)):
            profit = prices[i] - bestBuy
            bestProfit = max(bestProfit, profit)
            bestBuy = min(bestBuy, prices[i])
            
        return bestProfit